src/main/java/com/almostreliable/lazierae2/network/sync/handler/CustomDataHandler.java

```diff
@@ -0,0 +1,36 @@
+package com.almostreliable.lazierae2.network.sync.handler;
+
+import com.almostreliable.lazierae2.network.sync.DataHandler;
+import com.almostreliable.lazierae2.network.sync.IMenuSyncable;
+import net.minecraft.network.FriendlyByteBuf;
+
+import java.util.function.Supplier;
+
+public final class CustomDataHandler extends DataHandler<IMenuSyncable> {
```

Comment:
Delegating the methods to the `CustomDataHandler` allows saving the old value so `hasChanged()` can check if the data handler requires syncing.
Directly implementing the interface in `SideConfiguration` wouldn't pass the old value and therefor, it can't check if sync is required.